src/extensions/nodejsMigrationHelper.ts

```diff
@@ -0,0 +1,75 @@
+import { FirebaseError } from "../error";
+import * as extensionsApi from "./extensionsApi";
+import { logPrefix } from "./extensionsHelper";
+import { promptOnce } from "../prompt";
+import * as utils from "../utils";
+
+const changeMsg =
+  "Node.js 8 has been deprecated. It’s recommended to update this extension to ensure it is running on Node.js 10.";
+const billingMsgUpdate =
+  "This update includes an upgrade to Node.js 10 from Node.js 8, which is no longer maintained. Your project must be on the Blaze (pay as you go) plan to deploy Node.js 10 functions. Starting with this update, you will be charged a small amount when you deploy this extension, including when you make configuration changes and apply future updates.";
+const billingMsgCreate =
+  "This extension includes Node.js 10 functions. Your project must be on the Blaze (pay as you go) plan to deploy Node.js 10 functions. You will be charged a small amount when you deploy this extension, including when you make configuration changes and apply future updates.";
+
+function hasRuntime(resource: extensionsApi.Resource, runtime: string): boolean {
+  let resourceRuntime = "nodejs8"; // default to nodejs8 runtime
+  if (resource.properties && resource.properties.runtime) {
+    resourceRuntime = resource.properties.runtime;
+  }
+  return resourceRuntime == runtime;
+}
+
+function shouldDisplayMsg(
+  newSpec: extensionsApi.ExtensionSpec,
+  curSpec?: extensionsApi.ExtensionSpec
+): boolean {
+  const newResources = newSpec.resources;
+  const curResources = curSpec?.resources || [];
+  return (
+    newResources.some((r) => hasRuntime(r, "nodejs10")) &&
```

Comment:
This statement is a bit dense, and I think this would be a bit more readable if we split it into a couple variables:
```
const newResourcesUseNode10 = newResources.some((r) => hasRuntime(r, "nodejs10"));
const curResourcesUseNode8 = curResources.some((r) => hasRuntime(r, "nodejs8"));
return (newResourcesUseNode10 && (!curSpec || curResourcesUseNode8))
```

Reply:
Agreed!